Author: David Leonard <david.leonard@opengear.com>
Date:   Tue Aug 16 06:51:36 2016 +1000

    OG-3543 udhcpc: warn when dropping bad TLVs
    
    If the DHCP server sends us what appears to be a malformed (overlong)
    suboption, warn on stderr before dropping it.
    
    This should help one of our customers debug their custom DHCP server.
Index: busybox-1.31.0/networking/udhcp/dhcpc.c
===================================================================
--- busybox-1.31.0.orig/networking/udhcp/dhcpc.c
+++ busybox-1.31.0/networking/udhcp/dhcpc.c
@@ -481,6 +481,9 @@ tlv_putenv(uint8_t *option, const char *
 			break;
 		}
 		if (&p[OPT_DATA] + p[OPT_LEN] > optend) {
+			uint8_t maxlen = optend - &p[OPT_DATA];
+			bb_perror_msg("overlong TLV %s.%u (%u > %u), ignoring",
+				opt_name, p[OPT_CODE], p[OPT_LEN], maxlen);
 			break;
 		}
 		snprintf(name, sizeof name, "%s_%d", opt_name, p[OPT_CODE]);
